sirepo/package_data/static/js/silas.js

```diff
@@ -106,7 +110,14 @@ SIREPO.app.controller('SourceController', function (appState, frameCache, persis
              ? 'parameter' : '3d';
     };
 
+    const updateHasOldData = (dataIsOld) => {
+        appState.models.hasOldData = dataIsOld;
+        appState.saveChanges('hasOldData');
+    };
```

Comment:
after discussing with @moellep it seemed to make sense to just rerun the laserpulse sim when `$on('laserpulse.changed')` 
re: "Discard changes to example, change a crystal parameter, then run simulation: the plots don't show because hasOldData is still true. Subsequent runs work." I am not sure what you mean here. This is only for the laser pulse tab and not for the beamline tab

Reply:
I mean a laser pulse parameter, sorry about that...